grades/api.py

```diff
@@ -230,3 +232,33 @@ def generate_program_certificate(user, program):
         user.username,
         program.title
     )
+
+
+def update_combined_final_grade(user, course):
+    """
+    Update or create CombineFinalGrade
+    Args:
+        user (User): a django User
+        course (Course): a course model object
+
+    Returns:
+        None
+    """
+    mmtrack = get_mmtrack(user, course.program)
+    final_grade = mmtrack.get_best_final_grade_for_course(course)
+    if final_grade is None:
+        log.error('User [%s] does not have a final for course [%s]', user, course)
+        return
+
+    best_exam = mmtrack.get_best_proctored_exam_grade(course)
+    if best_exam is None:
+        log.error('User [%s] does not have a passing exam grade for course [%s]', user, course)
+        return
+
+    calculated_grade = round(final_grade.grade_percent * COURSE_GRADE_WEIGHT + best_exam.score * EXAM_GRADE_WEIGHT, 1)
```

Comment:
It would probably be good to refactor this and the calculation in `dashboard/api.py` into a util function so we don't have to keep those implementations in sync. Although, it also might make sense to just merge your changes from the other PR into this one since it solves the duplicate code issue too.

Reply:
So in my other PR I am removing this calculation in `dashboard/api.py`. Do you still want me to refactor this into utils?